playback/exoplayer/src/main/kotlin/ExoPlayerBackend.kt

```diff
@@ -0,0 +1,109 @@
+package org.jellyfin.playback.exoplayer
+
+import android.content.Context
+import com.google.android.exoplayer2.C
+import com.google.android.exoplayer2.DefaultRenderersFactory
+import com.google.android.exoplayer2.ExoPlayer
+import com.google.android.exoplayer2.MediaItem
+import com.google.android.exoplayer2.PlaybackException
+import com.google.android.exoplayer2.Player
+import com.google.android.exoplayer2.video.VideoSize
+import org.jellyfin.playback.core.backend.BasePlayerBackend
+import org.jellyfin.playback.core.mediastream.MediaStream
+import org.jellyfin.playback.core.model.PlayState
+import org.jellyfin.playback.core.model.PositionInfo
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.ZERO
+import kotlin.time.Duration.Companion.milliseconds
+
+class ExoPlayerBackend(
+	private val context: Context,
+) : BasePlayerBackend() {
+	override fun getPositionInfo(): PositionInfo = PositionInfo(
+		active = exoPlayer.currentPosition.milliseconds,
+		buffer = exoPlayer.bufferedPosition.milliseconds,
+		duration = if (exoPlayer.duration == C.TIME_UNSET) ZERO else exoPlayer.duration.milliseconds,
+	)
+
+	private val exoPlayer by lazy {
+		val renderersFactory = DefaultRenderersFactory(context).apply {
+			setEnableDecoderFallback(true)
+			setExtensionRendererMode(DefaultRenderersFactory.EXTENSION_RENDERER_MODE_ON)
+		}
+		ExoPlayer.Builder(context, renderersFactory)
+			.build()
+			.also { player -> player.addListener(PlayerListener()) }
+	}
+
+	inner class PlayerListener : Player.Listener {
```

Comment:
More of a nit, is it worth pulling derivative classe(s) out into a dedicated file(s) early? 

The existing Exoplayer/VLC code organically grew and turned into a mixture of boilerplate and implementation, hence I'm wondering if it's worth separating before we add to it?

Reply:
Right now the ExoPlayer implementation is small enough that it doesn't matter. When the class grows it will likely get split into multiple files. I also believe large files aren't a huge problem, the fact that it isn't related to the file it's in is the problem. In the current player code we have pretty much everything in one file, the "ExoPlayerBackend" only contains the binding to ExoPlayer.